tests/feature/ExportDumpTest.php

```diff
@@ -128,4 +163,173 @@ public function createsStreamedFileDownloadResponse()
         $this->assertInstanceOf(StreamedResponse::class, $response);
         $this->assertEquals(200, $response->getStatusCode());
     }
+
+    /**
+     * @test
+     * @dataProvider provideForHasCorrectConfiguration
+     */
+    public function hasCorrectConfiguration(array $protectorOptions, array $expected): void
+    {
+        $this->configureProtector($protectorOptions);
+
+        $connection = DB::connection($this->protector->getConnectionName());
+        $schemaState = $connection->getSchemaState();
+        $schemaStateProxy = $this->runProtectedMethod('getProxyForSchemaState', [$schemaState]);
+
+        $conditionalParameters = $schemaStateProxy->getConditionalParameters();
+
+        $this->assertEquals($expected[$connection->getDriverName()], $conditionalParameters);
+    }
+
+    public static function provideForHasCorrectConfiguration(): array
+    {
+        return [
+            'all options on' => [
+                'protectorOptions' => [
+                    self::PROTECTOR_WITH_CREATE_DB,
+                    self::PROTECTOR_WITH_DROP_DB,
+                    self::PROTECTOR_WITH_COMMENTS,
+                    self::PROTECTOR_WITH_CHARSETS,
+                    self::PROTECTOR_WITH_DATA,
+                    self::PROTECTOR_WITH_TABLESPACES,
+                ],
+                'expected' => self::getExpected([
+                    'pgsql' => [
```

Comment:
its a bit sad that this is basically never executed and actually tested

Reply:
We have a pull request for this which is not fully complete yet https://github.com/cybex-gmbh/laravel-protector/pull/93